server/src/main/java/org/apache/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -115,7 +116,7 @@ public CoordinatorDynamicConfig(
       @JsonProperty("mergeSegmentsLimit") int mergeSegmentsLimit,
       @JsonProperty("maxSegmentsToMove") int maxSegmentsToMove,
       @Deprecated @JsonProperty("percentOfSegmentsToConsiderPerMove") @Nullable Double percentOfSegmentsToConsiderPerMove,
-      @JsonProperty("useBatchedSegmentSampler") boolean useBatchedSegmentSampler,
+      @Deprecated @JsonProperty("useBatchedSegmentSampler") boolean useBatchedSegmentSampler,
```

Comment:
is setting the default value of `useBatchedSegmentSampler` to `true` missing?

Reply:
Thanks for catching this! It was in a different patch. 
Updated to use the default of `true` both when
- deserializing using the constructor. This would mean that configs already stored in the DB with no explicit value of `useBatchedSegmentSampler` would now start using `true`.
- creating/deserializing using the Builder. So newly created/updated configs would now  use `true`.

Also updated tests to verify this.